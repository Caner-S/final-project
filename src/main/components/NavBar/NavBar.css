.NavUl {
    list-style-type: none;
    margin: 0;
    padding-bottom: 3%;
    overflow: hidden;
    background-color: transparent;
}

.NavLi {
    float: left;
}

.NavLi a {
    display: block;
    color: #12ABDB;
    text-align: center;
    padding: 14px 16px;
    text-decoration: none;
}

.NavLi a:hover {
    background-color: lightgray;
;
}
