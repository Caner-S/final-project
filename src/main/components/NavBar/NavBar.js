import React, {Component} from "react";
import './NavBar.css';
import firebase from "../../../config/firebase";



class NavBar extends Component {


    render() {
        const renderAuthButton = ()=>{
            if(this.props.auth){
                return <li className="NavLi" style={{float: 'right'}}><a href="admin">Admin</a></li>
            }
        }
        return (
            <ul className="NavUl">
                <li className="NavLi"><a href="/">Home</a></li>
                <li className="NavLi"><a href="mybookings">My Bookings</a></li>
                <li className="NavLi"><a href="stats">Stats</a></li>
                <li className="NavLi" style={{float: 'right'}}><a onClick={() => firebase.auth().signOut()}>Sign Out</a></li>
                {renderAuthButton()}
            </ul>
        );
    }
}

export default NavBar;