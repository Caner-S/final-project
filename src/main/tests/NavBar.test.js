import React from 'react';
import renderer from 'react-test-renderer';

import NavBar from "../components/NavBar/NavBar";

it('renders correctly when there are no items', () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
});