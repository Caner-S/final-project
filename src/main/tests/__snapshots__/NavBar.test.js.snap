// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<ul
  className="NavUl"
>
  <li
    className="NavLi"
  >
    <a
      href="/"
    >
      Home
    </a>
  </li>
  <li
    className="NavLi"
  >
    <a
      href="mybookings"
    >
      My Bookings
    </a>
  </li>
  <li
    className="NavLi"
  >
    <a
      href="stats"
    >
      Stats
    </a>
  </li>
  <li
    className="NavLi"
    style={
      Object {
        "float": "right",
      }
    }
  >
    <a
      onClick={[Function]}
    >
      Sign Out
    </a>
  </li>
</ul>
`;
